llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2481,6 +2556,12 @@ void VPlanTransforms::convertToConcreteRecipes(VPlan &Plan,
       VPI->replaceAllUsesWith(VectorStep);
       ToRemove.push_back(VPI);
     }
+    for (VPRecipeBase &R : make_early_inc_range(*VPBB)) {
+      if (auto *ExtRed = dyn_cast<VPExtendedReductionRecipe>(&R))
+        expandVPExtendedReduction(ExtRed);
```

Comment:
```suggestion
      if (auto *ExtRed = dyn_cast<VPExtendedReductionRecipe>(&R)) {
        expandVPExtendedReduction(ExtRed);
        continue;
      }
```

otherwise there may be a use-after-free below if R has been removed.

Reply:
Updated, thanks.